### PR TITLE
Issue #SB-30124 fix: Enable "Create New" functionality for editors

### PR DESCRIPTION
**Pull request by vaibhavbhuva**

### CODE REVIEW

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]
There is no unit-test cases for create new functionality..

File: src/app/editors/collection-editor/collection-editor.component.ts [hunk 1]
```diff
@@ -1,22 +1,173 @@
...
+         console.log('Fetching content details failed ::', error);
+      });
   }
+
   editorEventListener(event) {
     this.router.navigate(['editors/content-list/collection-editor']);
+  }
+
+  private getContentDetails() {
+    if (this.queryParams.identifier) {
```
Comment (vinukumar-vs) [inline src/app/editors/collection-editor/collection-editor.component.ts:34]: This if condition should go ngOnInit function.

File: src/app/editors/collection-editor/collection-editor.component.ts [hunk 2]
```diff
@@ -1,22 +1,173 @@
...
+    } else {
+      return of({});
+    }
+  }
+
+  getChannel(channelId) {
+    this.helperService.getChannel(channelId)
+      .subscribe((response) => {
+        this.channelData = _.get(response, 'result.channel');
+        this.getFrameWorkDetails();
```
Comment (vinukumar-vs) [inline src/app/editors/collection-editor/collection-editor.component.ts:46]: Why this is called here in sequence & why are we waiting till the channel response?

File: src/app/editors/collection-editor/collection-editor.component.ts [hunk 3]
```diff
@@ -1,22 +1,173 @@
...
+      }, (error) => {
+        console.log(error);
+      });
+
+  }
+  getFrameWorkDetails() {
+    this.helperService.getCategoryDefinition('Collection', this.content.primaryCategory, this.content.channel)
```
Comment (vinukumar-vs) [inline src/app/editors/collection-editor/collection-editor.component.ts:53]: Use Rxjs to combine the requests in parallel or sequence.

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]
Use Rxjs for API request to combine or to call parallel

**Comment by vaibhavbhuva** [comment]
@vinukumar-vs All the comments have been resolved.

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**